lwt/httpaf_lwt.mli

```diff
@@ -1,24 +1,48 @@
 open Httpaf
 
+module Buffer : sig
+  type t
+
+  val create : int -> t
+
+  val get : t -> f:(Bigstringaf.t -> off:int -> len:int -> int) -> int
+  val put : t -> f:(Bigstringaf.t -> off:int -> len:int -> int Lwt.t) -> int Lwt.t
+end
+
+module type IO = sig
+  type t
+
+  val read : t -> Buffer.t -> [> `Eof | `Ok of int ] Lwt.t
+
+  val writev
+     : t
+    -> Faraday.bigstring Faraday.iovec list
+    -> [ `Closed | `Ok of int ] Lwt.t
+
+  val shutdown_send : t -> unit
+
+  val shutdown_receive : t -> unit
+
+  val close : t -> unit Lwt.t
+end
 
 (* The function that results from [create_connection_handler] should be passed
    to [Lwt_io.establish_server_with_client_socket]. For an example, see
    [examples/lwt_echo_server.ml]. *)
-module Server : sig
+module Server (Io: IO) : sig
   val create_connection_handler
     :  ?config         : Config.t
-    -> request_handler : (Unix.sockaddr -> Server_connection.request_handler)
-    -> error_handler   : (Unix.sockaddr -> Server_connection.error_handler)
-    -> Unix.sockaddr
-    -> Lwt_unix.file_descr
+    -> request_handler : (Io.t -> Server_connection.request_handler)
```

Comment:
This should take a `sockaddr`, not an `Io.t`. Doing so implies that users are free to use the socket, which they are not. It also puts the Async and lwt API's out of sync.

Reply:
This makes sense and I agree, however the pending question that led me to choose this signature in the first place is:

Mirage doesn’t use the Unix bits, so the type for request handler would probably also have to be parametrized over. Are you comfortable with that additional layer of parameterization? Cc @dinosaure who helped me with this yesterday

Reply:
Adding an `’addr` type parameter makes sense here.

Reply:
My trouble with it is that we can't tell that the Unix runtime would take `'addr -> request_handler` whereas the Mirage runtime would take `request_handler`. Not without parameterizing on `request_handler` at least (I think).

Reply:
See updated comment about putting it in `Io`.